cmd/oidc-broker/daemon/daemon.go

```diff
@@ -44,31 +45,35 @@ type daemonConfig struct {
 }
 
 // New registers commands and return a new App.
-func New() *App {
-	a := App{ready: make(chan struct{})}
+func New(name string) *App {
+	if name == "" {
```

Comment:
I would directly change to not make it optional then, and remove the global name altogether (to ensure we don’t use it by accident)